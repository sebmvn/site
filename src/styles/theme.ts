export default {
  grid: {
    container: '130rem',
    box: '94rem',
    gutter: '3.2rem'
  },
  border: {
    radius: '0.5rem'
  },
  font: {
    family:
      "Jost, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    muted: 200,
    light: 300,
    normal: 400,
    regular: 500,
    semibold: 700,
    bold: 800,
    black: 900,

    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '5.2rem'
    }
  },
  colors: {
    primary: '#0A7FF6',
    secondary: '#F6CE1F',

    mainBg: '#FFFFFF',
    secondaryBg: '#FCFCFC',
    headerBg: '#FCFCFC',

    menuText: '#636363',
    bannerLink: '#6E82FA',

    text: '#636363',
    secondaryText: '#FCFCFC',

    assets: '#0A7FF6',
    assets100: '#0A7FF6',

    inputBg: '#F4F4F4',
    cardBg: '#FFFFFF',

    btn: '#0A7FF6',
    btnBg: '#F6CE1F',
    btnLogin: '#0A7FF6',
    btnLoginBg: '#FFFFFF',

    progressBar: '#F4F4F4',

    iconBg: '#0A7FF6',
    footerBg: '#F4F4F4',
    copyrightBg: '#ECECEC',

    warning: '#CF0A1E',
    alert: '#EE1010',

    white: '#FFFFFF',
    black: '#000000',
    gray: '#8696A0',
    muted: 'rgba(0,0,0,0.6)'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
} as const

// const dark = {
//   primary: '#E9EDEF',
//   secondary: '#F6CE1F',

//   mainBg: '#141F27',
//   secondaryBg: '#202C33',
//   headerBg: '#141F27',

//   menuText: '#D1D7DB',
//   bannerLink: '#6E82FA',

//   text: '#D1D7DB',
//   secondaryText: '#D1D7DB',

//   assets: '#0A7FF6',
//   assets100: '#8696A0',

//   inputBg: '#2A3942',
//   cardBg: '#202C33',

//   btn: '#0A7FF6',
//   btnBg: '#F6CE1F',
//   btnLogin: '#F6CE1F',
//   btnLoginBg: '#2A3942',

//   progressBar: '#8696A0',

//   iconBg: '#D1D7DB',
//   footerBg: '#F4F4F4',
//   copyrightBg: '#ECECEC',

//   warning: '#CF0A1E',
//   alert: '#EE1010',
//   white: '#FFFFFF'
// }
