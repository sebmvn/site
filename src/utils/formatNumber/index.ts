export const formatPriceNotation = (value: string) => {
  value = value?.replace('.', '').replace(',', '').replace(/\D/g, '')

  if (isNaN(parseFloat(value))) {
    return ''
  }

  const options = { minimumFractionDigits: 2 }
  const result = new Intl.NumberFormat('pt-BR', options).format(
    parseFloat(value) / 100
  )

  return result
}

export const formatIntegerNotation = (value: string) => {
  return value?.replace('.', '').replace(',', '').replace(/\D/g, '')
}

export const toDecimal = (value: string) => {
  const formattedValue = formatIntegerNotation(value)
  return parseFloat(formattedValue!) / 100
}

export const toInteger = (value: string) => {
  const formattedValue = formatIntegerNotation(value)
  return parseInt(formattedValue!)
}
