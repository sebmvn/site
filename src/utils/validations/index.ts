import { CalendarType } from 'hooks/use-calc'
import Joi from 'joi'

const fieldsValidations = {
  initialValue: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),
  monthlyValue: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),

  interestRate: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),
  period: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),
  interestRateCalendar: Joi.string().valid('monthly', 'yearly').required(),
  periodCalendar: Joi.string().valid('monthly', 'yearly').required()
}

export type FieldErrors = {
  [key: string]: string
}

const getFieldErrors = (objError: Joi.ValidationResult) => {
  const errors: FieldErrors = {}

  if (objError.error) {
    objError.error.details.forEach((err) => {
      errors[err.path.join('.')] = err.message
    })
  }

  return errors
}

type compoundInterestCalculatorFields = {
  initialValue: string
  monthlyValue: string
  interestRate: string
  period: string
  interestRateCalendar: CalendarType
  periodCalendar: CalendarType
}
export const compoundInterestCalculatorValidate = (
  values: compoundInterestCalculatorFields
) => {
  const schema = Joi.object(fieldsValidations)

  return getFieldErrors(schema.validate(values, { abortEarly: false }))
}
