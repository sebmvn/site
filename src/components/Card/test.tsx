import theme from 'styles/theme'
import { renderWithTheme } from 'utils/test-utils'

import { Card } from '.'

describe('<Card />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <Card>
        <span>Hub do Investidor</span>
      </Card>
    )

    expect(container.firstChild).toHaveStyleRule(
      'background',
      theme.colors.mainBg
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
