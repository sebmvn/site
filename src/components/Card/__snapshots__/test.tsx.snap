// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background: #FFFFFF;
  padding: 3.2rem 2.4rem;
  border: 1px solid #F4F4F4;
  border-radius: 0.5rem;
  box-shadow: 0px 2px 4px #1932413F;
}

<div
  class="c0"
>
  <span>
    Hub do Investidor
  </span>
</div>
`;
